### PR TITLE
HOTFIX: Poll with zero milliseconds during restoration phase

### PR DESCRIPTION
**Pull request by guozhangwang**
1. After the poll call, re-check if the state has been changed or not; if yes, initialize the tasks again.
2. Minor log4j improvements.

### CODE REVIEW

**Comment by guozhangwang** [comment]
retest this please

**Comment by bbejeck** [comment]
@guozhangwang looks like the test failures are related.

**Comment by mjsax** [comment]
https://github.com/apache/kafka/pull/4085#issuecomment-338012466

File: streams/src/test/java/org/apache/kafka/streams/integration/AbstractResetIntegrationTest.java [hunk 1]
```diff
@@ -109,7 +109,11 @@ static void afterClassGlobalCleanup() {
     void beforePrepareTest() throws Exception {
         ++testNo;
         bootstrapServers = cluster.bootstrapServers();
-        mockTime = cluster.time;
+
+        // we align time to seconds to get clean window boundaries and thus ensure the same result for each run
+        // otherwise, input records could fall into different windows for different runs depending on the initial mock time
+        final long alignedTime = (System.currentTimeMillis() / 1000) * 1000;
+        mockTime = new MockTime(0, alignedTime, System.nanoTime());
```
Comment (mjsax) [inline streams/src/test/java/org/apache/kafka/streams/integration/AbstractResetIntegrationTest.java]: This seems to overlap with #4095 -- should not be part of this PR IMHO.

**Comment by dguy** [comment]
@guozhangwang are you also going to pause all partitions until the restoration is complete in this PR?

**Comment by bbejeck** [comment]
retest this please

**Comment by guozhangwang** [comment]
> @guozhangwang are you also going to pause all partitions until the restoration is complete in this PR?

Not in this PR. I plan to do that in another follow-up PR in https://issues.apache.org/jira/browse/KAFKA-6106.

File: streams/src/test/java/org/apache/kafka/streams/integration/AbstractResetIntegrationTest.java (hunk 1 above)
Comment (guozhangwang) [reply streams/src/test/java/org/apache/kafka/streams/integration/AbstractResetIntegrationTest.java]: Yup, it was included just trying to see if the Jenkins job would be more stable then. Will rebase once 4095 is merged in.

**Review by dguy — APPROVED** [review]
Thanks @guozhangwang, LGTM

**Comment by tedyu** [comment]
lgtm

**Comment by guozhangwang** [comment]
Merged to trunk and 1.0.
